### PR TITLE
feat: Add RevokeToken tool for removing saved payment methods

### PR DESCRIPTION
**Pull request by himanshu-rzp**
## Description

* Add RevokeToken function in tokens.go with customer_id and token_id parameters
* Register revoke_token tool in payments toolset as write tool
* Add comprehensive unit tests covering positive, negative, and edge cases
* Update README.md with new tool documentation and correct API link
* All tests pass and linter checks clean

## Type of Change

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] Refactoring (no functional changes, code improvements only)
* [ ] Documentation update
* [ ] Performance improvement

## Testing

* [x] Manual testing
* [x] Added unit tests
* [ ] Added integration tests (if applicable)
* [x] All tests pass locally

## Checklist

* [x] I have followed the code style of this project
* [x] I have added comments to code where necessary, particularly in hard-to-understand areas
* [x] I have updated the documentation where necessary
* [x] I have verified that my changes do not introduce new warnings or errors
* [x] I have checked for and resolved any merge conflicts
* [x] I have considered the performance implications of my changes

## Additional Information

The revoke_token tool allows users to cancel saved payment methods (tokens) for customers. This tool uses the Razorpay Cancel Token API endpoint with PUT method to properly revoke tokens from the payment system.

### CODE REVIEW

File: pkg/razorpay/tokens_test.go [hunk 1]
```diff
@@ -409,3 +409,255 @@ func Test_FetchSavedPaymentMethods_ClientContextScenarios(t *testing.T) {
...
+	t.Run("invalid client type in context", func(t *testing.T) {
+		// Create tool with nil client
+		tool := RevokeToken(obs, nil)
+
+		// Create context with invalid client type
+		ctx := contextkey.WithClient(context.Background(), "invalid_client_type")
+		request := mcpgo.CallToolRequest{
+			Arguments: map[string]interface{}{
+				"customer_id": "cust_1Aa00000000003",
+				"token_id":    "token_EhYXHrLsJdwRhM",
```
Comment (jating06) [inline pkg/razorpay/tokens_test.go]: change token with some dummy value

**Review by jating06 — APPROVED** [review]

**Review by jating06 — APPROVED** [review]
